### PR TITLE
sqlx-cli: init at 0.2.0

### PR DESCRIPTION
**Pull request by greizgh**
###### Motivation for this change

[sqlx-cli](https://github.com/launchbadge/sqlx/tree/master/sqlx-cli#sqlx-cli) is a development tool used in combination with sqlx library. It handles migrations and offline schema generation.

Checks are skipped because `cargo test` attempts to build all the workspaces, and one is failing.
([It doesn't seem possible for now to specify a cargo flag for tests.](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/default.nix#L247))

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/sqlx-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib, rustPlatform, fetchFromGitHub, pkgconfig, openssl }:
+
+rustPlatform.buildRustPackage {
+  pname = "sqlx-cli";
+  version = "0.2.0";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/sqlx-cli/default.nix:5]:
```suggestion
  version = "0.4.2";
```

File: pkgs/development/tools/rust/sqlx-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "launchbadge";
+    repo = "sqlx";
+    rev = "v0.4.2";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/sqlx-cli/default.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/development/tools/rust/sqlx-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 = "1q6p4qly9qjn333nj72sar6nbyclfdw9i9l6rnimswylj0rr9n27";
+  };
+
+  cargoSha256 = "1cxihnk82anyvryvl0d95323ppjn4v27ykz2c1sflmva7k34prhd";
+
+  doCheck = false;
+  cargoBuildFlags = [ "-p sqlx-cli" ];
+
+  nativeBuildInputs = [ pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/sqlx-cli/default.nix:19]:
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

File: pkgs/development/tools/rust/sqlx-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{ lib, rustPlatform, fetchFromGitHub, pkgconfig, openssl }:
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/sqlx-cli/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, pkg-config, openssl }:
```

File: pkgs/development/tools/rust/sqlx-cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ openssl ];
+
+  meta = with lib; {
+    description =
+      "SQLx's associated command-line utility for managing databases, migrations, and enabling offline mode with sqlx::query!() and friends.";
+    homepage = "https://github.com/launchbadge/sqlx";
+    license = licenses.asl20;
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/sqlx-cli/default.nix:27]:
```suggestion
```

set by buildRustPackage.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rust/sqlx-cli/default.nix (hunk 1 above)
Comment (greizgh) [reply pkgs/development/tools/rust/sqlx-cli/default.nix:5]: I'm confused since the CLI tool has a different version than the libs: https://github.com/launchbadge/sqlx/blob/master/sqlx-cli/Cargo.toml#L3
I think it makes more sense to keep the CLI version.

Comment (SuperSandro2000) [reply pkgs/development/tools/rust/sqlx-cli/default.nix:5]: The GitHub tags list 0.4.2 so the version is that. Doing it your way prevents ready updated and everyone thinks it is a mistake.

Comment (greizgh) [reply pkgs/development/tools/rust/sqlx-cli/default.nix]: OK it feels weird to have mismatching version, but if you are fine with it then :+1:

File: pkgs/development/tools/rust/sqlx-cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  meta = with lib; {
+    description =
+      "SQLx's associated command-line utility for managing databases, migrations, and enabling offline mode with sqlx::query!() and friends.";
+    homepage = "https://github.com/launchbadge/sqlx";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ greizgh ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/sqlx-cli/default.nix:27]: Please put this into a separate commit with the message ``maintainers: add greizgh``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rust/sqlx-cli/default.nix (hunk 6 above)
Comment (greizgh) [reply pkgs/development/tools/rust/sqlx-cli/default.nix:27]: There is already a separate commit to add add myself to the maintainer so I'm not sure what you mean here?
Do you want a commit *after* the one creating the package to append to the maintainers list?

Comment (SuperSandro2000) [reply pkgs/development/tools/rust/sqlx-cli/default.nix:27]: > There is already a separate commit to add add myself to the maintainer so I'm not sure what you mean here?

I exactly meant this.

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -3455,6 +3455,11 @@
       fingerprint = "7FC7 98AB 390E 1646 ED4D  8F1F 797F 6238 68CD 00C2";
     }];
   };
+  greizgh = {
+    email = "greizgh@ephax.org";
+    github = "greizgh";
+    githubId = 1313624;
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3461]:
```suggestion
    githubId = 1313624;
    name = "greizgh";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
